AElf.Consensus/ConsensusService.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using AElf.Common;
+using AElf.Kernel;
+using AElf.Kernel.Account;
+using AElf.Kernel.Services;
+using AElf.Kernel.Types;
+using AElf.Types.CSharp;
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace AElf.Consensus
+{
+    public class ConsensusService : IConsensusService
+    {
+        private readonly IConsensusObserver _consensusObserver;
+        private readonly IExecutingService _executingService;
+        private readonly IConsensusInformationGenerationService _consensusInformationGenerationService;
+        private readonly IAccountService _accountService;
+
+        private IDisposable _consensusObservables;
+
+        private byte[] _latestGeneratedConsensusInformation;
+
+        public ILogger<ConsensusService> Logger { get; set; }
+
+        public ConsensusService(IConsensusObserver consensusObserver, IExecutingService executingService,
+            IConsensusInformationGenerationService consensusInformationGenerationService,
+            IAccountService accountService)
+        {
+            _consensusObserver = consensusObserver;
+            _executingService = executingService;
+            _consensusInformationGenerationService = consensusInformationGenerationService;
+            _accountService = accountService;
+
+            Logger = NullLogger<ConsensusService>.Instance;
+        }
+
+        public async Task TriggerConsensus(int chainId)
+        {
+            var consensusCommand = ExecuteConsensusContract(chainId, await _accountService.GetAccountAsync(),
+                ConsensusMethod.GetConsensusCommand, Timestamp.FromDateTime(DateTime.UtcNow)).ToByteArray();
+
+            // Initial or update the schedule.
+            _consensusObservables?.Dispose();
```

Comment:
should not dispose here, you can throw exception when it is not disposed, but not dispose it.

if an object is disposed, the lifetime of a disposed object is finished.

if you keep a disposable object in a service, it is not acceptable cause it means the service is not stateless